### PR TITLE
[develop] before upgrade effect-asset

### PR DESCRIPTION
**Pull request by star-e**
Add custom-pipeline info to effect-asset.
This change will increase size of effect-asset.
Updated custom pipeline.

### Changelog

* added `pass` to `IPassStates`
* added `IDescriptorInfo`
* added `descriptors` to `IShaderInfo`

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.
  > Manual trigger with `@cocos-robot run test cases` afterward.
* [ ] does not change any runtime related code or build configuration
  > If any reviewer thinks the CI checks are needed, please uncheck this option, then close and reopen the issue.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

**Comment by star-e** [comment]
@cocos-robot run test cases

**Comment by star-e** [comment]
@cocos run test cases

File: cocos/asset/assets/effect-asset.ts [hunk 1]
```diff
@@ -143,6 +144,16 @@ export declare namespace EffectAsset {
...
+    export interface IDescriptorInfo {
+        rate: number;
+        blocks: IBlockInfo[];
+        samplerTextures: ISamplerTextureInfo[];
+        samplers: ISamplerInfo[];
+        textures: ITextureInfo[];
+        buffers: IBufferInfo[];
+        images: IImageInfo[];
+        subpassInputs: IInputAttachmentInfo[];
+    }
```
Comment (minggo) [inline cocos/asset/assets/effect-asset.ts:156]: Should change native too?

Comment (star-e) [reply cocos/asset/assets/effect-asset.ts:156]: Yes, I need to check this.

**Comment by star-e** [comment]
@cocos-robot run test cases

File: cocos/asset/assets/effect-asset.ts (hunk 1 above)
Comment (star-e) [reply cocos/asset/assets/effect-asset.ts:156]: Added native implementation.

**Review by VisualSJ — APPROVED** [review]

**Review by minggo — APPROVED** [review]

**Pull request merged**